generator/uscore/metadata_extractor.rb

```diff
@@ -330,23 +330,34 @@ def add_must_support_elements(profile_definition, sequence)
               end
             end
             sequence[:must_supports][:slices] << must_support_element
-            next
-          end
-          path = element['path'].gsub(sequence[:resource] + '.', '')
-          must_support_element = { path: path }
-          if element['fixedUri'].present?
-            must_support_element[:fixed_value] = element['fixedUri']
-          elsif element['patternCodeableConcept'].present?
-            must_support_element[:fixed_value] = element['patternCodeableConcept']['coding'].first['code']
-            must_support_element[:path] += '.coding.code'
-          elsif element['fixedCode'].present?
-            must_support_element[:fixed_value] = element['fixedCode']
-          elsif element['patternIdentifier'].present?
-            must_support_element[:fixed_value] = element['patternIdentifier']['system']
-            must_support_element[:path] += '.system'
+          elsif element['type'].any? { |type| type['code'] == 'Reference' }
+            reference_type = element['type'].find { |type| type['code'] == 'Reference' }
+            profiles = reference_type['targetProfile']
+            resource_types = profiles
+              .select { |profile| resources_by_type['StructureDefinition'].find { |stdef| stdef['url'] == profile } }
```

Comment:
This selects only the US Core references because those are the only profiles we have in `resources_by_type`?

Reply:
Right, all the structure definitions in the us core resource folder are us core profiles. I also can't filter by url because of the vital sign profiles which are base fhir.